File: lib/datamodel.java
public class Task {
    private String title;
    private String description;
    private String assignedVolunteer;
    private String dueDate;
    private String status;

    // Constructors, getters, and setters
    public Task() { }

    public Task(String title, String description, String assignedVolunteer, String dueDate, String status) {
        this.title = title;
        this.description = description;
        this.assignedVolunteer = assignedVolunteer;
        this.dueDate = dueDate;
        this.status = status;
    }

    // Getters and setters...
}

File: datamodel.dart
class Task {
  String title;
  String description;
  String assignedVolunteer;
  String dueDate;
  String status;

  Task(this.title, this.description, this.assignedVolunteer, this.dueDate, this.status);

  Task.fromJson(Map<String, dynamic> json)
      : title = json['title'],
        description = json['description'],
        assignedVolunteer = json['assignedVolunteer'],
        dueDate = json['dueDate'],
        status = json['status'];

  Map<String, dynamic> toJson() => {
    'title': title,
    'description': description,
    'assignedVolunteer': assignedVolunteer,
    'dueDate': dueDate,
    'status': status,
  };
}






